### PR TITLE
Jettify and update project structure

### PR DESCRIPTION
**Pull request by keithsmyth**
Updated from separate Android solutions to single Android solution, converted each step into a module, and then Jettified entire solution.

*I haven't changed any code directly, just build files

### CODE REVIEW

File: .idea/codeStyles/codeStyleConfig.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (benbaxter) [inline .idea/codeStyles/codeStyleConfig.xml]: Remove this file

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (benbaxter) [inline .idea/misc.xml]: Remove this

File: gradle.properties [hunk 3]
```diff
@@ -11,3 +11,7 @@ org.gradle.jvmargs=-Xmx1536m
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
+# Kotlin code style for this project: "official" or "obsolete":
```
Comment (benbaxter) [inline gradle.properties:14]: What is this sorcery?

File: local.properties [hunk 4]
```diff
@@ -0,0 +1,10 @@
+## This file is automatically generated by Android Studio.
+# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
+#
+# This file should *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+sdk.dir=/Users/keithsmyth/workspace/android/sdk
```
Comment (benbaxter) [inline local.properties]: This is not going to work for other people without your account on their computer.

File: local.properties [hunk 5]
```diff
@@ -0,0 +1,10 @@
+## This file is automatically generated by Android Studio.
```
Comment (benbaxter) [inline local.properties]: Remove this file

File: resizing-chromeos.iml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (benbaxter) [inline resizing-chromeos.iml]: Remove this file

File: step00-start/.gitignore [hunk 7]
```diff
@@ -1,10 +1 @@
-*.iml
-.gradle
-/local.properties
-/.idea/libraries
-/.idea/modules.xml
-/.idea/workspace.xml
-.DS_Store
 /build
```
Comment (benbaxter) [inline step00-start/.gitignore]: *.iml	
.gradle	
/local.properties	
/.idea/libraries	
/.idea/modules.xml	
/.idea/workspace.xml	
.DS_Store	
/build	/build
/captures	
.externalNativeBuild

File: step00-start/ASSETS_LICENSE.md [hunk 8]
```diff
@@ -1,101 +0,0 @@
-# Asset Licenses
```
Comment (benbaxter) [inline step00-start/ASSETS_LICENSE.md:1]: You need this file at the root project level

File: local.properties [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
+#
+# This file should *NOT* be checked into Version Control Systems,
```
Comment (dh--) [inline local.properties]: Look like this file should *NOT* be checked into Version Control

File: step00-start/CONTRIBUTING.md [hunk 10]
```diff
@@ -1,23 +0,0 @@
-# How to Contribute
```
Comment (benbaxter) [inline step00-start/CONTRIBUTING.md:1]: You need this at the root project level

File: step00-start/LICENSE [hunk 11]
```diff
@@ -1,202 +0,0 @@
-
```
Comment (benbaxter) [inline step00-start/LICENSE:1]: You need this at the root project level

File: step00-start/README.md [hunk 12]
```diff
@@ -1,36 +0,0 @@
-# Android App Resizing for Chrome OS
```
Comment (benbaxter) [inline step00-start/README.md:1]: You need this at the root project level

File: step00-start/.gitignore [hunk 13]
```diff
@@ -1,10 +1 @@
-*.iml
```
Comment (dh--) [inline step00-start/.gitignore]: Why remove all of these files from the .gitignore? These look like good files to exclude.

File: step01-viewmodel/build.gradle [hunk 14]
```diff
@@ -13,30 +13,41 @@
...
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+dependencies {
+    implementation fileTree(include: ['*.jar'], dir: 'libs')
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0-alpha01'
+    implementation 'androidx.recyclerview:recyclerview:1.1.0-alpha01'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha3'
+    //implementation files('libs/constraint-debug.aar')
```
Comment (benbaxter) [inline step01-viewmodel/build.gradle]: Is this needed? If not, remove it.

**Review by dh-- — COMMENTED** [review]
Looks pretty good -- but I think there are a few files that should have been excluded from the change list. ptal at the gitgnore files across the board.

File: step02-savedinstance/build.gradle [hunk 15]
```diff
@@ -13,30 +13,41 @@
...
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+dependencies {
+    implementation fileTree(include: ['*.jar'], dir: 'libs')
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0-alpha01'
+    implementation 'androidx.recyclerview:recyclerview:1.1.0-alpha01'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha3'
+    //implementation files('libs/constraint-debug.aar')
```
Comment (benbaxter) [inline step02-savedinstance/build.gradle]: If not needed, remove

File: step03-multiplelayouts/build.gradle [hunk 16]
```diff
@@ -13,30 +13,41 @@
...
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+dependencies {
+    implementation fileTree(include: ['*.jar'], dir: 'libs')
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0-alpha01'
+    implementation 'androidx.recyclerview:recyclerview:1.1.0-alpha01'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha3'
+    //implementation files('libs/constraint-debug.aar')
```
Comment (benbaxter) [inline step03-multiplelayouts/build.gradle]: If not needed, remove.

File: step00-start/.gitignore (hunk 13 above)
Comment (keithsmyth) [reply step00-start/.gitignore]: I accidentally copied over them when moving the modules, will restore

**Review by benbaxter — CHANGES_REQUESTED** [review]

File: step01-viewmodel/build.gradle (hunk 14 above)
Comment (keithsmyth) [reply step01-viewmodel/build.gradle]: Copied over, this will go in the next bug I address

File: local.properties (hunk 4 above)
Comment (keithsmyth) [reply local.properties]: arg, I overwrote some gitignore files by mistake, this is a side effect of that

File: gradle.properties (hunk 3 above)
Comment (keithsmyth) [reply gradle.properties:14]: Some kind of jetifier thing perhaps? It's the only thing that would have touched this file

File: step00-start/LICENSE (hunk 11 above)
Comment (keithsmyth) [reply step00-start/LICENSE:1]: Already there

File: step00-start/README.md (hunk 12 above)
Comment (keithsmyth) [reply step00-start/README.md:1]: Already there

File: step01-viewmodel/build.gradle (hunk 14 above)
Comment (keithsmyth) [reply step01-viewmodel/build.gradle]: I ended up removing it because it won't play nice with jetpack

File: step04-animation/build.gradle [hunk 17]
```diff
@@ -44,8 +44,7 @@ dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'androidx.appcompat:appcompat:1.1.0-alpha01'
     implementation 'androidx.recyclerview:recyclerview:1.1.0-alpha01'
-    //    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
-    implementation files('libs/constraint-debug.aar')
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha3'
```
Comment (benbaxter) [inline step04-animation/build.gradle:47]: Do you need the alpha build? I think you said you did but I wanted to makes sure.

**Review by benbaxter — APPROVED** [review]

File: step04-animation/build.gradle (hunk 17 above)
Comment (keithsmyth) [reply step04-animation/build.gradle:47]: Correct, MotionLayout is coming and that will be a nice chunky delete delete delete PR :)

**Pull request merged**